tests/playbooks/integration_pytest_python3.yml

```diff
@@ -0,0 +1,104 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Install dependencies for integration tests
+  hosts: all
+  vars:
+    - rpmdependencies:
+        - git
+        - python3-pip
+        - rsync
+
+  tasks:
+    - name: Install EPEL for RHEL and CentOS
+      # yamllint disable-line rule:line-length
+      command: "yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm"
+      when: ansible_distribution in ["CentOS", "RedHat"]
+
+    - name: Install rpm dependencies
+      package:
+        state: present
+        name: "{{ rpmdependencies }}"
+
+    - name: Search for rsync path
+      command: which rsync
+
+    - name: Install Pytest
+      command: "pip3 install pytest"
+
+- name: Run Pytest tests
+  hosts: all
+  vars:
+    - rundir: /run/system-roles-test/
+  tasks:
+    - file:
+        state: directory
+        path: /run
+
+    - command: git rev-parse --show-toplevel
+      register: git_top_directory
+      delegate_to: localhost
+
+    - debug:
+        var: git_top_directory
+
+    - synchronize:
+        src: "{{ git_top_directory.stdout }}/"
+        dest: "{{ rundir }}/"
+        recursive: yes
+        delete: yes
+        rsync_opts:
+          - "--exclude=.pyc"
+          - "--exclude=__pycache__"
+      when: False
+
+    # TODO: using tar and copying the file is a workaround for the synchronize
+    # module that does not work in test-harness. Related issue:
+    # https://github.com/linux-system-roles/test-harness/issues/102
+    #
+    - name: Create Tar file
+      shell: 'tar -cvf {{ git_top_directory.stdout }}/testrepo.tar
+             --exclude "*.pyc" --exclude "__pycache__" --exclude testrepo.tar
+             -C {{ git_top_directory.stdout }} .'
+      delegate_to: localhost
+
+    - name: Copy testrepo.tar to the remote system
+      copy:
+        src: "{{ git_top_directory.stdout }}/testrepo.tar"
+        dest: "{{ rundir }}"
+
+    - name: Untar testrepo.tar
+      command: tar xf testrepo.tar
+      args:
+        chdir: "{{ rundir }}"
+
+
+# Import needed in order to install Network Manager dependencies in the remote
+# system.
+- import_playbook: "../tests_default_nm.yml"
+- hosts: all
+  vars:
+    - rundir: /run/system-roles-test/
+  tasks:
+    - block:
+        - name: Run pytest with nm
+          command: "pytest {{ rundir }}/tests/integration/ --provider=nm"
+          register: playbook_run
+      always:
+        - debug:
+            var: playbook_run.stdout_lines
+
+# Import needed in order to install initscripts dependencies into the remote
+# system.
+- import_playbook: "../tests_default_initscripts.yml"
```

Comment:
It would probably be nicer to move the imports for installing dependencies to the top right after installing the other dependencies, then all the task for running pytest can be in one playbook. Then the `rundir` variable does not need to be defined several times.